var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -69,31 +64,35 @@ class Hdf5(AutotoolsPackage):
     variant('java', default=False, description='Enable Java support')
     variant('threadsafe', default=False,
             description='Enable thread-safe capabilities')
-
+    variant('tools', default=False, description='Enable build tools')
     variant('mpi', default=True, description='Enable MPI support')
     variant('szip', default=False, description='Enable szip support')
+    variant('zlib', default=True, description='Enable zlib support')
     variant('pic', default=True,
             description='Produce position-independent code (for shared libs)')
-    # Build HDF5 with API compaitibility.
-    variant('api', default='none', description='choose api compatibility', values=('none', 'v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
+    # Build HDF5 with API compatibility.
+    variant('api', default='default', description='Choose api compatibility for earlier version', values=('default', 'v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
 
     conflicts('api=v114', when='@1.6:1.12.99', msg='v114 is not compatible with this release')
     conflicts('api=v112', when='@1.6:1.10.99', msg='v112 is not compatible with this release')
     conflicts('api=v110', when='@1.6:1.8.99', msg='v110 is not compatible with this release')
     conflicts('api=v18', when='@1.6:1.6.99', msg='v18 is not compatible with this release')
 
-    depends_on('autoconf', type='build', when='@develop')
-    depends_on('automake', type='build', when='@develop')
-    depends_on('libtool',  type='build', when='@develop')
-    depends_on('m4',       type='build', when='@develop')
+    depends_on('cmake@3.12:')
 
     depends_on('mpi', when='+mpi')
-    depends_on('java', when='+java')
+    depends_on('java', type=('build', 'run'), when='+java')
     # numactl does not currently build on darwin
     if sys.platform != 'darwin':
         depends_on('numactl', when='+mpi+fortran')
-    depends_on('szip', when='+szip')
-    depends_on('zlib@1.1.2:')
+    depends_on('libaec', when='+szip')
+    depends_on('zlib@1.2.5:', when='+zlib')
```

Comment:
@lrknox I'm sorry but why did you change the version constraint for `zlib`?

Reply:
Arbitrary modernization.  If you tell me it's problematic I'll keep it at 1.1.2:.

Regarding an earlier question about the zlib variant, I said we would take it out (and always include zlib), but without the variant, netcdf-c fails to build because it requires hdf5 with zlib and apparently checks the spec, so I left the zlib variant in.

Reply:
I would recommend not to change the version constraint unless absolutely necessary (and even then I would ask to check whether the constraint is unconditional, i.e. maybe the older versions of `hdf5` are happy with older versions of `zlib`).

Could the problem with the `zlib` variant be that you removed the variant but the dependency on `zlib` was still constrained with it? In other words, did you also remove `when='+zlib'` from the `depends_on('zlib')` directive when you removed the `variant('zlib')` directive?

Reply:
The problem with netcdf-c was that when removing the variant I also removed "self.define('HDF5_ENABLE_Z_LIB_SUPPORT', True)".  With that statement, and "depends_on('zlib@1.1.2:')", zlib is always included and the netcdf-c requirement for zlib is satisfied.